packages/core/src/utils/shell-utils.test.ts

```diff
@@ -420,6 +422,39 @@ describe('stripShellWrapper', () => {
   });
 });
 
+describe('isShellInvocationAllowlisted', () => {
+  const createInvocation = (command: string) =>
+    ({ params: { command } }) as unknown as AnyToolInvocation;
+
+  it('should return false when any chained command segment is not allowlisted', () => {
```

Comment:
can you add a specific test with a pipe

Reply:
Added an explicit pipeline scenario so each segment must be allowlisted; it fails without the new parsing.